tests/openapi/openapi_integration/test_query_full.py

```diff
@@ -0,0 +1,551 @@
+
+    response = request_with_validation(
+        api="/collections/{collection_name}/points/search",
+        method="POST",
+        path_params={"collection_name": collection_name},
+        body={
+            "vector": {
+                "name": "dense-text",
+                "vector": [0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8]
+            },
+            "filter": filter,
+            "limit": 10,
+        },
+    )
+    assert response.ok
+    search_result_2 = response.json()["result"]
+
+    response = request_with_validation(
+        api="/collections/{collection_name}/points/search",
+        method="POST",
+        path_params={"collection_name": collection_name},
+        body={
+            "vector": {
+                "name": "sparse-text",
+                "vector": {
+                    "indices": [63, 65, 66],
+                    "values": [1, 2.2, 3.3],
+                }
+            },
+            "filter": filter,
+            "limit": 10,
+        },
+    )
+    assert response.ok
+    search_result_3 = response.json()["result"]
+
+    response = request_with_validation(
+        api="/collections/{collection_name}/points/search",
+        method="POST",
+        path_params={"collection_name": collection_name},
+        body={
+            "vector": {
+                "name": "dense-multi",
+                "vector": [3.05, 3.61, 3.76, 3.74], # legacy API expands single vector to multiple vectors
+            },
+            "filter": filter,
+            "limit": 10,
+        },
+    )
+    assert response.ok
+    search_result_4 = response.json()["result"]
+
+    rrf_expected = reciprocal_rank_fusion([search_result_1, search_result_2, search_result_3, search_result_4], limit=10)
+
+    response = request_with_validation(
+        api="/collections/{collection_name}/points/query",
+        method="POST",
+        path_params={"collection_name": collection_name},
+        body={
+            "prefetch": [
+                {
+                    "query": [0.1, 0.2, 0.3, 0.4],
+                    "using": "dense-image"
+                },
+                {
+                    "query": [0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8],
+                    "using": "dense-text"
+                },
+                {
+                    "query": {
+                        "indices": [63, 65, 66],
+                        "values": [1, 2.2, 3.3],
+                    },
+                    "using": "sparse-text",
+                },
+                {
+                    "query": [[3.05, 3.61, 3.76, 3.74]],
+                    "using": "dense-multi"
+                },
+            ],
+            "filter": filter,
+            "limit": 10,
+            "query": {"fusion": "rrf"}
+        },
+    )
+    assert response.ok, response.json()
+    rrf_result = response.json()["result"]
+
+    def get_id(x):
+        return x["id"]
+
+    # rrf order is not deterministic with same scores, so we need to sort by id
```

Comment:
Non-related to this PR: We could make it deterministic by using `ScoredPointTies`, but do we want to?